src/vs/code/electron-main/windows.ts

```diff
@@ -441,7 +441,17 @@ export class WindowsManager implements IWindowsMainService {
 		}
 
 		// Handle empty
-		if (emptyToOpen.length > 0) {
+		if (emptyToRestore.length > 0) {
+			emptyToRestore.forEach(backupFolder => {
+				const configuration = this.toConfiguration(openConfig, null, null, null, null, backupFolder);
```

Comment:
@Tyriar something is weird about `backupFolder`. As someone that tries to understand the code I would assume every window has an associated `backupFolder`. but that is not true, only the empty windows to restore have it. I would make this more obvious. Maybe remove the property from `IWindowConfiguration` and just pass it into `openInBrowserWindow` method? Also needs a clearer name to make its meaning more obvious.